src/Features/Core/Navigation/IDocumentNavigationService.cs

```diff
@@ -25,16 +25,16 @@ internal interface IDocumentNavigationService : IWorkspaceService
         /// <summary>
         /// Navigates to the given position in the specified document, opening it if necessary.
         /// </summary>
-        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool usePreviewTab = false);
+        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, bool reopenDocument = false, bool usePreviewTab = false);
```

Comment:
So what's the guidance for this flag? If I'm writing a new feature, how do I choose the value? Should the "real fix" here be that if the file is already open we don't go through a full-open path and just a lighter weight piece of code to transition the tab state?